src/orders/components/OrderRefundFulfilledProducts/OrderRefundFulfilledProducts.tsx

```diff
@@ -0,0 +1,244 @@
+import CardContent from "@material-ui/core/CardContent";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+import { CSSProperties } from "@material-ui/styles";
+import CardTitle from "@saleor/components/CardTitle";
+import Money from "@saleor/components/Money";
+import Skeleton from "@saleor/components/Skeleton";
+import TableCellAvatar from "@saleor/components/TableCellAvatar";
+import { FormsetChange } from "@saleor/hooks/useFormset";
+import { renderCollection } from "@saleor/misc";
+import { OrderRefundData_order_fulfillments } from "@saleor/orders/types/OrderRefundData";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { OrderRefundFormData } from "../OrderRefundPage/form";
+
+const useStyles = makeStyles(
+  theme => {
+    const inputPadding: CSSProperties = {
+      paddingBottom: theme.spacing(2),
+      paddingTop: theme.spacing(2)
+    };
+
+    return {
+      cartContent: {
+        paddingBottom: 0,
+        paddingTop: 0
+      },
+      colQuantity: {
+        textAlign: "right",
+        width: 210
+      },
+      notice: {
+        marginBottom: theme.spacing(1),
+        marginTop: theme.spacing(2)
+      },
+      orderNumber: {
+        display: "inline",
+        marginLeft: theme.spacing(1)
+      },
+      quantityInnerInput: {
+        ...inputPadding
+      },
+      quantityInnerInputNoRemaining: {
+        paddingRight: 0
+      },
+      remainingQuantity: {
+        ...inputPadding,
+        color: theme.palette.text.secondary,
+        whiteSpace: "nowrap"
+      },
+      setMaximalQuantityButton: {
+        marginTop: theme.spacing(1)
+      }
+    };
+  },
+  { name: "OrderRefundFulfilledProducts" }
+);
+
+interface OrderRefundFulfilledProductsProps {
+  fulfillment: OrderRefundData_order_fulfillments;
+  data: OrderRefundFormData;
+  disabled: boolean;
+  orderNumber: string;
+  onRefundedProductQuantityChange: FormsetChange<string>;
+  onSetMaximalQuantities: () => void;
+}
+
+const OrderRefundFulfilledProducts: React.FC<OrderRefundFulfilledProductsProps> = props => {
+  const {
+    fulfillment,
+    data,
+    disabled,
+    orderNumber,
+    onRefundedProductQuantityChange,
+    onSetMaximalQuantities
+  } = props;
+  const classes = useStyles(props);
+  const intl = useIntl();
+
+  return (
+    <Card>
+      <CardTitle
+        title={
+          <>
+            {intl.formatMessage({
+              defaultMessage: "Fulfillment",
+              description: "section header"
+            })}
+            <Typography className={classes.orderNumber} variant="body1">
+              {`#${orderNumber}-${fulfillment?.fulfillmentOrder}`}
```

Comment:
maybe we should have some placeholder(or hide all text) when fulfillment is undefined?

Reply:
yep, I've changed it to hiding if fulfillment is undefined